### ISSUE TITLE
Issue with Swagger behind reverse proxy pointing to part of a service

### ISSUE THREAD

**Issue by raphlau** [domaindrivendev/Swashbuckle#1251]
## VERSION: ##
5.6.0

## STEPS TO REPRODUCE: ##
The setup is relatively simple (WebApi project):
- Swagger is located at root.com/api/swagger/docs/{apiVersion}:
```C#
 config.EnableSwagger(
                    "api/swagger/docs/{apiVersion}",
[...]
```
- SwaggerUI is at root.com/api/swagger/index:
```C#
.EnableSwaggerUi("api/swagger/{*assetPath}", swagger => [...]
```
- The website itself is behind a reverse proxy:
`proxy.com/myservice` -> `root.com/api`

- `RootUrl` is set up for the reverse proxy

Problem: If I try hitting `proxy.com/myservice/swagger/index`, I am able to see the UI. However, it tries to load the JSON from `root.com/api/api/swagger/docs/v1` instead of `root.com/api/swagger/docs/v1`. It seems that what I need is to be able to change the `routeTemplate` dynamically, or tell the UI where the JSON file is in a dynamic way.

I tried using a `CustomProvider` (`ISwaggerProvider`), but that doesn't seem to be used to resolve the path of the JSON itself, only the services.

Basically, because my reverse proxy is pointing to `/api`, swagger needs to be located at `/api/something`. However, doing so via the `routeTemplate` makes the swagger UI try to load `/api/api/something`, probably because it reuses the `routeTemplate`.

Can we add support for this workflow?

**Comment by shaunparsloe** [comment]
I am pretty sure that can be sorted with the code in #705  
Make sure that your proxy sets the X-Forwarded-* headers

**Comment by raphlau** [comment]
The issue you linked gives a way to obtain the originating host. I don't have a problem with that, the issue I'm facing happens even if I hard code the values to the correct ones for the specific case I'm trying to cover.
